### PR TITLE
Configuration: add initial NonStop values in OpenSSL::config

### LINKED ISSUE #12972: Configure cannot guess the NonStop system automatically
With the help of [OpenSSL::config](https://github.com/openssl/openssl/blob/master/util/perl/OpenSSL/config.pm), `./Configure` now guesses all system flavors (even Windows) automatically from standard settings.  I know that for NonStop, there's the complication of different threading models, but maybe there's a possibility to do a good enough selection based on `uname` data, it would be a start.

@rsbecker, perhaps you have some ideas of "standard" host triplets that could be used to select at least some of the OpenSSL configs?  If you have `config.guess`, its answer in diverse environments could be a starting point.

### PR DESCRIPTION
**Pull request by levitte**
This makes Configure work it's automatic config detection, at least for
the simple straightfoward cases.

Fixes #12972

### CODE REVIEW

**Review by beldmit — APPROVED** [review]
If works for @rsbeckerca

**Comment by rsbeckerca** [comment]
Let's change it so that if `NSV` is reported, it forces `nsx` instead.

File: util/perl/OpenSSL/config.pm
Comment (rsbeckerca) [inline util/perl/OpenSSL/config.pm:165]: -> `'nsv-tandem-nsk${RELEASE}'`

File: util/perl/OpenSSL/config.pm
Comment (rsbeckerca) [inline util/perl/OpenSSL/config.pm:876]: -> `{ target => 'nonstop-nsv' }`

**Review by rsbeckerca — CHANGES_REQUESTED** [review]
Please see my comments. The NSV and NSX platforms use the same binaries, so NSV is redundant.

File: util/perl/OpenSSL/config.pm
Comment (levitte) [reply util/perl/OpenSSL/config.pm:165]: This translates uname data to standard host triplets.  I'd rather keep those here.  However, the next place is where we convert the host triplet to OpenSSL config names, that's the proper spot for this kind of enforcement.

File: util/perl/OpenSSL/config.pm
Comment (levitte) [reply util/perl/OpenSSL/config.pm:876]: Sure.

Comment (levitte) [reply util/perl/OpenSSL/config.pm:876]: Done

**Comment by rsbeckerca** [comment]
This looks good.

**Review by rsbeckerca — APPROVED** [review]

**Review by beldmit — DISMISSED** [review]
LGTM, works for the customer.

**Comment by rsbeckerca** [comment]
Sadly, the current commit, once merged with everything else, does not work. I'm looking into it.

File: util/perl/OpenSSL/config.pm
Comment (rsbeckerca) [inline util/perl/OpenSSL/config.pm:878]: Should be
      [ 'NSE-L-.*-NONSTOP_KERNEL.*', { target => 'nonstop-nse' } ],

File: util/perl/OpenSSL/config.pm
Comment (rsbeckerca) [inline util/perl/OpenSSL/config.pm:879]: Should be
      [ 'NSV-L-.*-NONSTOP_KERNEL.*', { target => 'nonstop-nsv' } ],

File: util/perl/OpenSSL/config.pm
Comment (rsbeckerca) [inline util/perl/OpenSSL/config.pm:880]: Should be
`      [ 'NSX-L-.*-NONSTOP_KERNEL.*', { target => 'nonstop-nsx' } ],`

**Review by rsbeckerca — COMMENTED** [review]
Please see my comments in the latest code.

File: util/perl/OpenSSL/config.pm
Comment (rsbeckerca) [reply util/perl/OpenSSL/config.pm:880]: I think I have the changes correct in comments. The matching string on the `nonstop-nse` platform is: `NSE-L-whatever-NONSTOP_KERNEL`. This decomposes into `NSE-L`, the detailed model where `NSE` is significant, and obviously `NONSTOP_KERNEL` is a nonstop.

**Comment by mspncp** [comment]
@levitte there is a typo in the commit message. Can be fixed during merge. 

> straightfo **r**!ward

File: util/perl/OpenSSL/config.pm
Comment (levitte) [reply util/perl/OpenSSL/config.pm:878]: The error was actually higher up...  I had switched around the interpretation of uname values.  Just pushed a fixup

**Comment by levitte** [comment]
> @levitte there is a typo in the commit message. Can be fixed during merge.

Fixed :wink:

**Comment by levitte** [comment]
@rsbecker, if you wouldn't mind trying again?  All I need to know is that `./Configure` guesses the system correctly

**Comment by rsbeckerca** [comment]
> 
> 
> @rsbecker, if you wouldn't mind trying again? All I need to know is that `./Configure` guesses the system correctly

Yes, it did indeed.

**Comment by levitte** [comment]
Ping

**Comment by rsbeckerca** [comment]
@levitte I'm still good with this. It guesses correctly on both NonStop platforms with defaults.

**Comment by levitte** [comment]
Cool, so it's now just up to @openssl/committers to approve :wink:

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**
